docs/guides/Guide-Headers.md

```diff
@@ -136,3 +136,5 @@ class EditInfoScreen extends React.Component {
 
 
 To see the rest of the header options, see the [navigation options document](/docs/navigators/navigation-options#Stack-Navigation-Options).
+
+As an alternative to `setParams`, you may want to consider using a state management library such as MobX or Redux, and when navigating to a screen, pass an object which contains the data necessary for the screen to render, as well as functions you may want to call that modify the data, make network requests and etc. That way, both your screen component and the static `navbarOptions` block will have access to the object. When following this approach, make sure to consider deep linking, which works best in cases where only javascript primitives are passed as navigation props to your screen. In case when deep linking is necessary, you may use a HOC to transform the primitives to the object your screen components expects.
```

Comment:
NIt: Beginners might not be aware of what a HOC is. Can you change "HOC" to "Higher Order Component (HOC)" and include a link to the React docs? https://reactjs.org/docs/higher-order-components.html 😄